### PR TITLE
Get browser list with PowerShell if reg query fails

### PR DESCRIPTION
**Pull request by alexey-lin**

### CODE REVIEW

File: src/utils/promisify.js [hunk 1]
```diff
@@ -2,6 +2,6 @@ import Promise from 'pinkie';
 import pify from 'pify';
 
 
-export default function (fn) {
-    return pify(fn, Promise);
+export default function (fn, multiArgs) {
```
Comment (AndreyBelym) [inline src/utils/promisify.js]: I think it's better to use the native `Promise` and `util.promisify`.

`child_process.exec` has a special native promisified version, which resolves `{ stdout, stderr }` as a child process results. It's perfect for our use case/

File: src/api/get-installations.js [hunk 2]
```diff
@@ -29,32 +30,71 @@ async function addInstallation (installations, name, instPath) {
...
 
 async function detectMicrosoftEdge () {
-    var regKey = 'HKCU\\Software\\Classes\\ActivatableClasses';
-    var stdout = await execWinShellUtf8(`@echo off & reg query ${regKey} /s /f MicrosoftEdge /k && echo SUCCESS || echo FAIL`);
+    const regKey = 'HKCU\\Software\\Classes\\ActivatableClasses';
+    const stderr = (await execWinShellUtf8(`@echo off & reg query ${regKey} /s /f MicrosoftEdge /k`, true))[2];
 
-    return /SUCCESS/.test(stdout) ? ALIASES['edge'] : null;
+    if (stderr) {
+        if (!/Registry editing has been disabled/i.test(stderr))
```
Comment (kirovboris) [inline src/api/get-installations.js]:
```js
const registryEditingDisabledRe = /Registry editing has been disabled/i;
```
and so on.

File: src/api/get-installations.js [hunk 3]
```diff
@@ -29,32 +30,71 @@ async function addInstallation (installations, name, instPath) {
...
+        const output = spawnSync('powershell.exe', ['-NoLogo', '-NonInteractive', '-Command', `Get-ChildItem -Path Registry::${regKey} -Recurse`]);
+
+        if (!/\bMicrosoftEdge\b/i.test(output.stdout.toString()))
+            return null;
+    }
+
+    return ALIASES['edge'];
 }
 
 async function searchInRegistry (registryRoot) {
```
Comment (kirovboris) [inline src/api/get-installations.js]: neat: let's extract two approach searching of installed browsers into functions, this method is too huge to read now.

File: src/utils/promisify.js (hunk 1 above)
Comment (alexey-lin) [reply src/utils/promisify.js]: As we decide to always use PowerShell, we don't need this anymore.

File: src/api/get-installations.js (hunk 2 above)
Comment (alexey-lin) [reply src/api/get-installations.js]: Already done.

File: src/api/get-installations.js (hunk 3 above)
Comment (alexey-lin) [reply src/api/get-installations.js]: Let's try to abandon the old approach.

**Comment by alexey-lin** [comment]
FPR

**Pull request merged**